applications/nrf_desktop/src/modules/hid_forward.c

```diff
@@ -1187,81 +1228,87 @@ static void hogp_out_rep_write_cb(struct bt_hogp *hogp, struct bt_hogp_rep_info
 
 	/* Send update if it was queued. */
 	uint8_t sent_report_id = bt_hogp_rep_id(rep);
-	sys_snode_t *cur_node;
-	sys_snode_t *prev_node = NULL;
+	struct subscriber *sub = get_subscriber(per);
 
-	SYS_SLIST_FOR_EACH_NODE(&per->enqueued_out_reports, cur_node) {
-		struct enqueued_out_report *cur_report = CONTAINER_OF(cur_node,
-								      struct enqueued_out_report,
-								      node);
+	if (!(per->enqueued_out_reports_bm & BIT(sent_report_id))) {
+		return;
+	}
 
-		uint8_t cur_report_id = cur_report->data[0];
+	for (size_t i = 0; i < ARRAY_SIZE(sub->out_reports); i++) {
+		if ((sub->saved_out_reports_bm & BIT(i)) &&
```

Comment:
Maybe we could even assert that `(sub->saved_out_reports_bm & BIT(sent_report_id)` if the `per->enqueued_out_reports_bm & BIT(sent_report_id)`? That would reduce number of checks for release configurations and we are sure that this must be true.